src/main/scala/it/pagopa/interop/authorizationprocess/api/impl/ClientApiServiceImpl.scala

```diff
@@ -265,14 +262,15 @@ final case class ClientApiServiceImpl(
     logger.info(operationLabel)
 
     val result: Future[Keys] = for {
-      userId         <- getUidFutureUUID(contexts)
-      clientUuid     <- clientId.toFutureUUID
-      organizationId <- getOrganizationIdFutureUUID(contexts)
-      client         <- authorizationManagementService
+      userUUID      <- getUidFutureUUID(contexts)
+      clientUuid    <- clientId.toFutureUUID
+      selfcareUUID  <- getSelfcareIdFutureUUID(contexts)
+      requesterUUID <- getOrganizationIdFutureUUID(contexts)
+      client        <- authorizationManagementService
         .getClient(clientUuid)
-        .ensureOr(client => OrganizationNotAllowedOnClient(clientId, client.consumerId))(_.consumerId == organizationId)
-      relationshipId <- securityOperatorRelationship(client.consumerId, userId).map(_.id)
-      seeds = keysSeeds.map(_.toDependency(relationshipId, dateTimeSupplier.get()))
+        .ensureOr(client => OrganizationNotAllowedOnClient(clientId, client.consumerId))(_.consumerId == requesterUUID)
+      user          <- getSecurityUser(selfcareUUID, client.consumerId, userUUID)
```

Comment:
Here we just need to check that the user is in the client
```suggestion
      _          <- client.users.find(_ == userUUID).toFuture(UserNotFound)
```